apps/meteor/app/api/server/v1/rooms.ts

```diff
@@ -280,25 +287,38 @@ API.v1.addRoute(
 			if (!(await canAccessRoomIdAsync(this.urlParams.rid, this.userId))) {
 				return API.v1.unauthorized();
 			}
-
-			const file = await Uploads.findOneById(this.urlParams.fileId);
-
-			if (!file) {
+			const filesarray: Partial<IUpload>[] = [];
```

Comment:
The API media confirm should not be changed to support multiple files. Please update the sendMessage API https://github.com/RocketChat/Rocket.Chat/blob/develop/apps/meteor/app/lib/server/methods/sendMessage.ts#L119 and any other API to send messages to accept a list of file ids to confirm

Reply:
Sir, I have added a commit that updates the sendMessage API to accept a list of file IDs. This allows us to share and confirm multiple files within a single message. You can review the changes in the latest commit. Please let me know if any further modifications are needed.